include/seqan3/range/decorator/gap_decorator_anchor_set.hpp

```diff
@@ -0,0 +1,830 @@
+    //!\brief Use the size_type of the underlying sequence.
+    using size_type = size_type_t<inner_type>;
+    //!\brief Use the difference_type of the underlying sequence.
+    using difference_type = difference_type_t<inner_type>;
+    //!\brief The iterator type of this container (a bidirectional iterator).
+    using iterator = gap_decorator_anchor_set_iterator;
+    //!\brief The const_iterator equals the iterator type. Since no references are ever returned and thus the underlying
+    //!        sequence cannot be modified through the iterator there is no need for const.
+    using const_iterator = iterator;
+    //!\}
+
+    /*!\name Constructors, destructor and assignment.
+     * \{
+     */
+    //!\brief Default constructor.
+    constexpr gap_decorator_anchor_set() = default;
+    //!\brief Copy constructor.
+    constexpr gap_decorator_anchor_set(gap_decorator_anchor_set const &) = default;
+    //!\brief Copy construction via assignment.
+    constexpr gap_decorator_anchor_set & operator=(gap_decorator_anchor_set const &) = default;
+    //!\brief Move constructor.
+    constexpr gap_decorator_anchor_set(gap_decorator_anchor_set && rhs) = default;
+    //!\brief Move assignment.
+    constexpr gap_decorator_anchor_set & operator=(gap_decorator_anchor_set && rhs) = default;
+    //!\brief Use default deconstructor.
+    ~gap_decorator_anchor_set() = default;
+
+    //!\brief Construct by another ungapped view.
+    constexpr gap_decorator_anchor_set(inner_type view) : ungapped_view{std::move(view)} {}
+
+    //!\brief Construct by another ungapped range by wrapping it into a seqan3::view::all.
+    template <typename range_type>
+    //!\cond
+        requires !std::is_same_v<range_type, gap_decorator_anchor_set> &&
+                 !std::ranges::View<range_type> &&
+                 std::ranges::RandomAccessRange<range_type>
+    //!\endcond
+    constexpr gap_decorator_anchor_set(range_type const & range) : gap_decorator_anchor_set(view::all(range)) {}
+    //!\}
+
+    /*!\brief Returns the total length of the aligned sequence.
+     * \returns The total length of the aligned sequence (gaps included).
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    size_type size() const noexcept
+    {
+        if (anchors.size())
+            return (*(anchors.rbegin())).second + ungapped_view.size();
+
+        return ungapped_view.size();
+    }
+
+    /*!\name Aligned sequence modifications
+     * \{
+     */
+    /*!\brief Insert a gap of length count at the aligned sequence iterator position.
+     * \param it     Iterator indicating the gap start position in the aligned sequence.
+     * \param count  Number of gap symbols to be inserted.
+     * \returns      An iterator pointing to the start position of the insertion.
+     *
+     * ### Complexity
+     *
+     * Average and worst case (insertion before last gap): \f$o(k)\f$,
```

Comment:
Are you sure? This was written by Marie and I have not put a lot of thought into the runtime analysis yet

Reply:
In the main docs in this file, we write O(k). I think o(k) wouldn't hold, since if we have 0 or 1 gaps, we don't have the update penalty of k-1, and for o(k) it would have to hold for all k >= 0, whereas for O(k) there is a number x such that it hold for all k >= x.